nipype/interfaces/niftyreg/regutils.py

```diff
@@ -0,0 +1,739 @@
+                            requires=['warp_files'],
+                            desc=desc)
+
+    desc = 'Average images and demean average image that have linear and \
+non-rigid transformations to a common space'
+    demean3_ref_file = File(position=1,
+                            argstr='-demean3 %s',
+                            xor=['avg_files', 'avg_lts_files', 'avg_ref_file',
+                                 'demean1_ref_file', 'demean2_ref_file'],
+                            requires=['warp_files'],
+                            desc=desc)
+
+    desc = 'transformation files and floating image pairs/triplets to the \
+reference space'
+    warp_files = traits.List(File(exist=True),
+                             position=-1,
+                             argstr='%s',
+                             sep=' ',
+                             xor=['avg_files', 'avg_lts_files'],
+                             desc=desc)
+
+    out_file = File(genfile=True,
+                    position=0,
+                    desc='Output file name',
+                    argstr='%s')
+
+
+class RegAverageOutputSpec(TraitedSpec):
+    """ Output Spec for RegAverage. """
+    out_file = File(desc='Output file name')
+
+
+class RegAverage(NiftyRegCommand):
+    """Interface for executable reg_average from NiftyReg platform.
+
+    Compute average matrix or image from a list of matrices or image.
+    The tool can be use to resample images given input transformation
+    parametrisation as well as to demean transformations in Euclidean or
+    log-Euclidean space.
+
+    This interface is different than the others in the way that the options
+    will be written in a command file that is given as a parameter.
+
+    For source code, see https://cmiclab.cs.ucl.ac.uk/mmodat/niftyreg
+
+    Examples
+    --------
+    >>> from nipype.interfaces.niftyreg import RegAverage
+    >>> node = RegAverage()  # doctest: +SKIP
+    >>> one_file = 'im1.nii'
+    >>> two_file = 'im2.nii'
+    >>> three_file = 'im3.nii'
+    >>> node.inputs.avg_files = [one_file, two_file, three_file]  \
+    # doctest: +SKIP
+    >>> node.cmdline  # doctest: +SKIP
+    'reg_average --cmd_file reg_average_cmd'
+
+    """
+    _cmd = get_custom_path('reg_average')
+    input_spec = RegAverageInputSpec
+    output_spec = RegAverageOutputSpec
+    _suffix = 'avg_out'
+
+    def _gen_filename(self, name):
+        if name == 'out_file':
+            if isdefined(self.inputs.avg_lts_files):
+                return self._gen_fname(self._suffix, ext='.txt')
+            elif isdefined(self.inputs.avg_files):
+                _, _, ext = split_filename(self.inputs.avg_files[0])
+                if ext not in ['.nii', '.nii.gz', '.hdr', '.img', '.img.gz']:
+                    return self._gen_fname(self._suffix, ext=ext)
+            return self._gen_fname(self._suffix, ext='.nii.gz')
+        return None
+
+    @property
+    def cmdline(self):
```

Comment:
what is the idea behind this? I have the impression that this should be just an option of the input spec.